dart/utils/PackageResourceRetriever.cpp

```diff
@@ -71,32 +71,37 @@ void PackageResourceRetriever::addPackageDirectory(
 }
 
 //==============================================================================
-bool PackageResourceRetriever::exists(const std::string& _uri)
+bool PackageResourceRetriever::exists(const common::Uri& _uri) const
 {
   std::string packageName, relativePath;
   if (!resolvePackageUri(_uri, packageName, relativePath))
     return false;
 
-  for(const std::string &packagePath : getPackagePaths(packageName))
+  for(const std::string& packagePath : getPackagePaths(packageName))
   {
-    const std::string localUri = "file://" + packagePath + relativePath;
-    if (mLocalRetriever->exists(localUri))
+    common::Uri fileUri;
+    fileUri.fromPath(packagePath + relativePath);
```

Comment:
If we use `fromRelativeUri` unlike the original implementation, then this should be like:
```cpp
common::Uri fileUri;
fileUri.fromRelativeUri(common::Uri::createFromPath(packagePath), relativePath);
```

But I realized that there is an issue. Since `relativePath` always have `/` at the front by the nature of path component of URI, `fromRelativeUri` regards it as an absolute path. So the resolved path always set to the `relativePath` without the base URI's path.